### ISSUE TITLE
Reporting a new crash

### ISSUE THREAD

**Issue by carlosduclos** [hashicorp/terraform#8594]
This happened under similar conditions to #8572.

This time it happened on a different resource, but the crash seems very similar. No network issues occurred at the time of the crash.

Our configuration is split among several files (10 separate files), in order to keep control of the different resources.

### Terraform Configuration Files
Cannot paste the configuration files, but I can send them privately if requested.

### Panic Output
https://gist.github.com/carlosduclos/4fa783a97e617b54ffcc884d136067de

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform destroy -target=some_resource`

### Important Factoids
Running on azure using ARM.

### References
#8572

**Comment by carlosduclos** [comment]
Interestingly enough, the crash happened because the resource being deleted had changed name. Shouldn't this be handled a little more graciously?

**Comment by stack72** [comment]
Morning @carlosduclos 

ok, so this is really weird. You seem to be hitting a case where the response from Azure is empty. This shouldn't be the case at all. The code is as follows:

```
resp, err := vmClient.Get(resGroup, name, "")
	if resp.StatusCode == http.StatusNotFound {
		d.SetId("")
		return nil
	}
```

Whether the name of the VM has changed or the VM has even been deleted, we should *always* have a response from the API and a response should always have a statusCode. 

The interested line above your panic is this one:

```
2016/09/01 07:21:55 [DEBUG] Request to https://management.azure.com/subscriptions/<redacted>/resourceGroups/si-northeurope-resource-group/providers/Microsoft.Compute/virtualMachines/si-api?api-version=2016-03-30 completed with no response
```

I am not sure what is causing Azure to not give you a response right here. I don't believe there are any conditions when a request to an API doesn't return any level of response - that would be against what HTTP does. A response would always have at least a statuscode in it

Is this a repeatable error? Can you try and run terraform with something like wireshark to see if it helps diagnose the issue? 

Paul

**Comment by carlosduclos** [comment]
I can try to repeat this, unfortunately it does not happen all of the time. Interestingly enough, terraform never crashed before for us until we upgraded to 0.7.2. The reason for the upgrade was the availability of network peering.

**Comment by stack72** [comment]
Hey @carlosduclos 

So that is even weirder - we have the following tests running nightly

https://travis-ci.org/hashicorp/terraform/builds/156709760

So I am pretty confident if there was an error in the library then something would show up - that's what is so bizarre in this case

P.

**Comment by carlosduclos** [comment]
I had the same thing happen to me again today. Terraform crashed several times both during creation and destruction, even when planning.
I installed wireshark and I waited until it started happening. I made sure that the crash was reproducable by repeating the instructions several times and then I enable wireshark. As usual, there were no traces of the crash while wireshark was running.
I got the same issue later today, and the same thing happened again. This leads me to think that there might be some weird timing issue somewhere.
Our infrastructure resides in North Europe (Azure datacenter is located in Amsterdam), however I'm on a business trip in India so there is some latency in the communication. I will be back in Europe next week, and I'll check if this happens again.
